loco-new/base_template/src/controllers/auth.rs

```diff
@@ -145,6 +150,60 @@ async fn current(auth: auth::JWT, State(ctx): State<AppContext>) -> Result<Respo
     format::json(CurrentResponse::new(&user))
 }
 
+/// Magic link authentication provides a secure and passwordless way to log in to the application.
+///
+/// # Flow
+/// 1. **Request a Magic Link**:  
+///    A registered user sends a POST request to `/magic-link` with their email.  
+///    If the email exists, a short-lived, one-time-use token is generated and sent to the user's email.  
+///    For security and to avoid exposing whether an email exists, the response always returns 200, even if the email is invalid.
+///
+/// 2. **Click the Magic Link**:  
+///    The user clicks the link (/magic-link/:token), which validates the token and its expiration.  
+///    If valid, the server generates a JWT and responds with a [`LoginResponse`].  
+///    If invalid or expired, an unauthorized response is returned.
+///
+/// This flow enhances security by avoiding traditional passwords and providing a seamless login experience.
+async fn magic_link(
+    State(ctx): State<AppContext>,
+    Json(params): Json<MagicLinkParams>,
+) -> Result<Response> {
+    let Ok(user) = users::Model::find_by_email(&ctx.db, &params.email).await else {
+        // we don't want to expose our users email. if the email is invalid we still
+        // returning success to the caller
+        tracing::debug!(email = params.email, "user not found by email");
+        return format::empty_json();
+    };
+
+    let user = user.into_active_model().create_magic_link(&ctx.db).await?;
```

Comment:
a user overwriting a user with `create_magic_link` is a bit confusing
maybe 
let mut user 
user.create_magic_link(..);

because actually it sets the magic link on the user.